projects/plugins/jetpack/modules/google-fonts/class-jetpack-google-fonts.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Module: Jetpack Google Fonts
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Manages Google fonts registration, introspection and enqueueing.
+ */
+class Jetpack_Google_Fonts {
+	/**
+	 * The Google fonts that we're registered after
+	 * running the `jetpack_google_fonts_list`.
+	 *
+	 * @var array
+	 */
+	private $registered_google_fonts = array();
+
+	/**
+	 * Hook into WordPress to register
+	 * and enqueue webfonts.
+	 */
+	public function __construct() {
+		$this->register_google_fonts_provider();
+
+		add_action( 'after_setup_theme', array( $this, 'register_google_fonts' ) );
+
+		/**
+		 * We are already enqueueing all registered fonts by default when loading the block editor,
+		 * so we only need to scan for webfonts when browsing as a guest.
+		 */
+		if ( ! is_admin() ) {
+			add_action( 'wp_loaded', array( $this, 'enqueue_global_styles_google_fonts' ) );
+			add_filter( 'pre_render_block', array( $this, 'scan_block_for_google_fonts' ), 10, 2 );
+		}
```

Comment:
Need to keep it like that until Gutenberg lazy loads font-face declarations as they're picked in the block editor.